src/main/java/app/nzyme/core/rest/resources/dot11/BanditsResource.java

```diff
@@ -0,0 +1,277 @@
+import jakarta.validation.constraints.NotNull;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.SecurityContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Path("/api/dot11/bandits")
+@Produces(MediaType.APPLICATION_JSON)
+public class BanditsResource extends UserAuthenticatedResource {
+
+    private static final Logger LOG = LogManager.getLogger(BanditsResource.class);
+
+    @Inject
+    private NzymeNode nzyme;
+
+    @GET
+    @RESTSecured(value = PermissionLevel.ANY, featurePermissions = { "dot11_monitoring_manage" })
+    @Path("/builtin")
+    public Response findAllBuiltIn() {
+        List<BuiltinBanditDetailsResponse> bandits = Lists.newArrayList();
+
+        for (Dot11BanditDescription bandit : Dot11Bandits.BUILT_IN) {
+            bandits.add(BuiltinBanditDetailsResponse.create(
+                    bandit.id(),
+                    bandit.name(),
+                    bandit.description(),
+                    bandit.fingerprints() == null ? Collections.emptyList() : bandit.fingerprints()
+            ));
+        }
+
+        return Response.ok(bandits).build();
+    }
+
+    @GET
+    @RESTSecured(value = PermissionLevel.ANY, featurePermissions = { "dot11_monitoring_manage" })
+    @Path("/builtin/show/{id}")
+    public Response findOneBuiltIn(@PathParam("id") @NotEmpty String id) {
+        Dot11BanditDescription bandit = null;
+        for (Dot11BanditDescription b : Dot11Bandits.BUILT_IN) {
+            if (b.id().equals(id)) {
+                bandit = b;
+                break;
+            }
+        }
+
+        if (bandit == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        return Response.ok(BuiltinBanditDetailsResponse.create(
+                bandit.id(),
+                bandit.name(),
+                bandit.description(),
+                bandit.fingerprints() == null ? Collections.emptyList() : bandit.fingerprints()
+        )).build();
+    }
+
+    @GET
+    @RESTSecured(value = PermissionLevel.ANY, featurePermissions = { "dot11_monitoring_manage" })
+    @Path("/custom")
+    public Response findAllCustom(@Context SecurityContext sc,
+                                  @QueryParam("limit") int limit,
+                                  @QueryParam("offset") int offset,
+                                  @QueryParam("organization_uuid") @NotNull UUID organizationId,
+                                  @QueryParam("tenant_uuid") @NotNull UUID tenantId) {
+        AuthenticatedUser authenticatedUser = getAuthenticatedUser(sc);
+
+        if (limit > 250) {
+            LOG.warn("Requested limit larger than 250. Not allowed.");
+            return Response.status(Response.Status.UNAUTHORIZED).build();
```

Comment:
Do you think a BAD_REQUEST here would be a bit more appropriate?

Reply:
:+1:

Reply:
Fixed in 9ca29d47